IPython/core/ultratb.py

```diff
@@ -248,6 +248,7 @@ def fix_frame_records_filenames(records):
         # Look inside the frame's globals dictionary for __file__,
         # which should be better. However, keep Cython filenames since
         # we prefer the source filenames over the compiled .so file.
+        filename = py3compat.cast_unicode(filename, "utf-8")
```

Comment:
Yep, you're right. This error doesn't happen in Python 3. Should I change to `cast_unicode_py2` then?